Gemfile

```diff
@@ -0,0 +1,60 @@
+source 'https://rubygems.org'
+
+
```

Comment:
This rarely actually matters. The order comes into play when bundler is loading the files, and sometimes weird ordering will cause rails to not boot. It's not a common problem.

So, yes, I definitely prefer more explicit groups, putting all global gems at the top then using a block to group gems in each of the four groups you listed together.